crates/exec-wasmtime/src/runtime/vfs/connect.rs

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! A file system containing listening networking streams.
+
+use super::super::net::tls;
+use super::super::WasiResult;
+
+use std::any::Any;
+use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4};
+use std::ops::Deref;
+use std::path::MAIN_SEPARATOR;
+use std::sync::Arc;
+
+use anyhow::Context;
+use cap_std::net::TcpStream;
+use enarx_config::{ConnectFile, FileName};
+use rustls::{Certificate, PrivateKey, RootCertStore};
+use wasi_common::file::{FdFlags, FileType};
+use wasi_common::{Error, ErrorExt, WasiDir, WasiFile};
+use wasmtime_vfs_dir::Directory;
+use wasmtime_vfs_ledger::InodeId;
+use wasmtime_vfs_memory::{Data, Inode, Link, Node};
+use wiggle::async_trait;
+use zeroize::Zeroizing;
+
+enum Socket {
+    Tcp(Link<()>),
+    Tls(Link<Arc<rustls::ClientConfig>>),
+}
+
+#[async_trait]
```

Comment:
Sounds like a good idea that should be discussed. We already have other occurences using this trait, so please file a ticket, because such change is out-of-scope for this PR.

Reply:
I've filed https://github.com/enarx/enarx/issues/2377 to track this.